minifi/minifi-bootstrap/src/test/java/org/apache/nifi/minifi/bootstrap/command/CommandRunnerFactoryTest.java

```diff
@@ -0,0 +1,155 @@
+ * limitations under the License.
+ */
+
+package org.apache.nifi.minifi.bootstrap.command;
+
+import static org.apache.nifi.minifi.bootstrap.BootstrapCommand.DUMP;
+import static org.apache.nifi.minifi.bootstrap.BootstrapCommand.ENV;
+import static org.apache.nifi.minifi.bootstrap.BootstrapCommand.FLOWSTATUS;
+import static org.apache.nifi.minifi.bootstrap.BootstrapCommand.RESTART;
+import static org.apache.nifi.minifi.bootstrap.BootstrapCommand.RUN;
+import static org.apache.nifi.minifi.bootstrap.BootstrapCommand.START;
+import static org.apache.nifi.minifi.bootstrap.BootstrapCommand.STATUS;
+import static org.apache.nifi.minifi.bootstrap.BootstrapCommand.STOP;
+import static org.apache.nifi.minifi.bootstrap.BootstrapCommand.UNKNOWN;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.verifyNoInteractions;
+
+import java.io.File;
+import org.apache.nifi.minifi.bootstrap.MiNiFiParameters;
+import org.apache.nifi.minifi.bootstrap.RunMiNiFi;
+import org.apache.nifi.minifi.bootstrap.service.BootstrapFileProvider;
+import org.apache.nifi.minifi.bootstrap.service.CurrentPortProvider;
+import org.apache.nifi.minifi.bootstrap.service.GracefulShutdownParameterProvider;
+import org.apache.nifi.minifi.bootstrap.service.MiNiFiCommandSender;
+import org.apache.nifi.minifi.bootstrap.service.MiNiFiExecCommandProvider;
+import org.apache.nifi.minifi.bootstrap.service.MiNiFiStatusProvider;
+import org.apache.nifi.minifi.bootstrap.service.MiNiFiStdLogHandler;
+import org.apache.nifi.minifi.bootstrap.service.PeriodicStatusReporterManager;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class CommandRunnerFactoryTest {
+
+    @Mock
+    private MiNiFiCommandSender miNiFiCommandSender;
+    @Mock
+    private CurrentPortProvider currentPortProvider;
+    @Mock
+    private MiNiFiParameters miNiFiParameters;
+    @Mock
+    private MiNiFiStatusProvider miNiFiStatusProvider;
+    @Mock
+    private PeriodicStatusReporterManager periodicStatusReporterManager;
+    @Mock
+    private BootstrapFileProvider bootstrapFileProvider;
+    @Mock
+    private MiNiFiStdLogHandler miNiFiStdLogHandler;
+    @Mock
+    private File bootstrapConfigFile;
+    @Mock
+    private RunMiNiFi runMiNiFi;
+    @Mock
+    private GracefulShutdownParameterProvider gracefulShutdownParameterProvider;
+    @Mock
+    private MiNiFiExecCommandProvider miNiFiExecCommandProvider;
+
+    @InjectMocks
+    private CommandRunnerFactory commandRunnerFactory;
+
+    @Test
+    void testRunCommandShouldStartCommandReturnStartRunner() {
+        CommandRunner runner = commandRunnerFactory.getRunner(START);
+
+        assertTrue(runner instanceof StartRunner);
```

Comment:
JUnit 5 now supports `assertInstanceOf`, which can be used here and in other methods.